### DISCUSSION TITLE
InventoryManager does not register an inv?

### DISCUSSION THREAD

**Discussion by dgrfps** [RyseInventory/RyseInventory#47] [General]
I'm wondering why InventoryManager declares setInventory as private, how am i supposed to register a inv in there?

`protected void setInventory(@NotNull UUID uuid,
                                @NotNull RyseInventory inventory) {
        this.inventories.put(uuid, inventory);
    }`
It also has getInv, but this is never registered, it keeps returning an error > java.util.NoSuchElementException: No value present.
 
` public @NotNull Optional<RyseInventory> getInventory(@NotNull UUID uuid) {`
    
After creating an inv, how can i access it from another method/class?
   ```
public static void RegisterAll() { 
        RyseInventory pInv = RyseInventory.builder().title("PARTICLES").rows(6).provider(new InventoryProvider() {
            @Override
            public void init(Player player, InventoryContents contents) {
            }
        }).identifier("particles").build(Main.getInstance());
}

@EventHandler
    public void Click(PlayerInteractEvent e) {
        if (e.getItem() == null) return;
        if (e.getAction() == Action.RIGHT_CLICK_BLOCK || e.getAction() == Action.RIGHT_CLICK_AIR)
        {
            e.setCancelled(true);
            Main.inventoryManager.getInventory("particles").get().open(e.getPlayer());  //particles not registered!?
        }
    }
```

**Comment by dgrfps** [comment]
I mean what is the purpose of getInventory?
I'll have to Declare a class var RyseInventory and then access it calling MyClass.inv right?

```
public class mInv implements Listener {
  public static RyseInventory inv;
  public static void createInv()
  {
   inv = RyseInventory.builder().title("PARTICLES").rows(6).provider(new InventoryProvider() {
         @Override
         public void init(Player player, InventoryContents contents) {
         }
     }).identifier("particles").build(Main.getInstance());
     
  }
  
  //CALLING FROM ANOTHER CLASS
@EventHandler
 public void Click(PlayerInteractEvent e) {
     if (e.getItem() == null) return;
     if (e.getAction() == Action.RIGHT_CLICK_BLOCK || e.getAction() == Action.RIGHT_CLICK_AIR)
     {
         e.setCancelled(true);
       //  Main.inventoryManager.getInventory("particles").get().open(e.getPlayer());  //particles not registered!?
       mInv.inv.open(e.getPlayer());
     }
 }
}
```

**Comment by Rysefoxx** [comment]
It won't work that way. It is not registered because it is not open by any player. Makes sense somehow.

I also don't quite understand what you're trying to do there. You can just create the inventory at the PlayerInteractEvent and open it there?

**Reply by dgrfps** [reply]
Yes, but should i call inv.open() inside `@Override public void init(Player player, InventoryContents contents) { } ` or after the .build(plugin) ?
I though i should create the inv only once, not inside every event, like, if i wanna access the same inv but from differents events/methods, what should i do?

**Reply by Rysefoxx** [reply]
You can only open the inventory after building.
You can cache the inventory and build only 1x.

Otherwise you get the inventory through different methods in the InventoryManager class. E.g. by a UUID. The player must have the inventory open, of course, otherwise what should he get? Caching all inventories 24/7 consumes RAM unnecessarily.

**Reply by Rysefoxx** [reply]
I could possibly add a builder option to cache it 24/7. Even if the inventory is not opened by anyone.

**Reply by dgrfps** [reply]
Right, i didn't know about this, i gonna cache the inv after building it
